### PR TITLE
Load path file from single location only

### PR DESCRIPTION
**Pull request by virtuald**
- ` /home/lvuser/py/deploy/pathplanner/*.path` on RoboRIO
- `__main__ / deploy/pathplanner/*.path` in simulation

### CODE REVIEW

File: pathplannerlib/src/sources.patch [hunk 1]
```diff
@@ -1,45 +1,56 @@
...
++
++static fs::path getDeployDirectory() {
++    static fs::path deployPath =
++#ifdef __FRC_ROBORIO__
++        fs::path(frc::filesystem::GetOperatingDirectory()) / "py" / "deploy" / "pathplanner";
++#else
++        getMainPath() / "deploy" / "pathplanner";
++#endif
++    return deployPath;
++}
```
Comment (TheTripleV) [inline pathplannerlib/src/sources.patch]: This, without the "pathplanner", should go in  robotpy-wpilib. It has the filesystem functions disabled.

**Review by TheTripleV — COMMENTED** [review]
Will paths relative to robot.py still work? We've been using relative paths for all of our extra files (pathplanner and pathweaver before it).

File: pathplannerlib/src/sources.patch (hunk 1 above)
Comment (virtuald) [reply pathplannerlib/src/sources.patch]: I suppose that's true.

**Comment by virtuald** [comment]
The path is relative to `dirname(__main__.__file__)`, which should generally work. I'll try it on a roborio later tonight, but there's no reason it shouldn't work AFAIK.

**Comment by virtuald** [comment]
Confirmed on RoboRIO that this approach tries to load from `/home/lvuser/py/deploy/pathplanner/*.path`

**Comment by virtuald** [comment]
Will depend on https://github.com/robotpy/robotpy-wpilib/pull/710 once merged.

**Comment by TheTripleV** [comment]
Paths relative to robot.py don't work with this PR.
A sample from our code::
```py
self.scheduler.schedule(commands.autonomous.DriveTrajectory(self, './paths/5Ball1', 2, 3, False))
```
`./paths/5Ball1` resolved before but doesn't now.

I get that pathplannerlib originally looked in a specific folder, but everywhere else in robot code, relative paths are relative to robot.py and it would be confusing to deviate from that.

**Comment by virtuald** [comment]
When you specify a robot 'project' in pathplanner, by default puts the files in the specific folder as you mention. It seems confusing to force users to have to copy the pathplanner files (as presumably your team does?). Additionally, the API specifies loading by name, not by a path, so this is staying with the original intent of the API.

With all that said, I agree that it would make sense to be able to load from any arbitrary location also, so maybe let's add another function that does that? Not sure what the right name would be though... `loadFile`?

**Comment by TheTripleV** [comment]
It's weird because here, the `Path` in `loadPath` refers to a trajectory but it generally refers to a `PathLike`. I'm fine with `loadFile`. Alternatively, would `loadPath` trying both be bad?

**Pull request merged**